### PR TITLE
Concepts inferrer

### PR DESCRIPTION
**Pull request by harrisonpim**
### What is it
A skeleton _inferrer_ which aggregates data about concepts from external sources

### What does it do
At the moment, it just looks for an `lc_names_id` in the library of congress names and wikidata APIs, and returns the relevant resulting data in a single, tidy response.

eg. hitting `/lc-names/nr91020770` gives [the response linked here](https://gist.githubusercontent.com/harrisonpim/553057ee5b50bf06d3facd0d24c74776/raw/83d83f56b7a4ac40ae396156ba45c526e2ec7505/concepts_lc_names_api_response.json) - an aggregation of the data exactly matching `nr91020770` in lc-names and wikidata.

### What's next
Trying to keep these PRs relatively small, but I'm planning to add endpoints for wikidata, LCSH and MeSH IDs, and include the intersections/overlaps of them in the responses. 

Also planning to do some inference/disambiguations where exact links don't exist between concepts on different platforms, but are similar enough to share useful data. eg even if there isn't an established link between the wikidata page for `leg` and the MeSH page for `leg`, if we can infer that they're referring to the same Concept, we should aggregate their data)

### CODE REVIEW

File: inference/app/src/exact/lc_names.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    preferred_label = None
+    for element in api_response:
+        if preferred_label_id in element:
+            preferred_label = element[preferred_label_id][0]["@value"]
+    return preferred_label
+
+
+def get_lc_names_variants(lc_names_id, api_response=None):
+    api_response = api_response or get_lc_names_api_response(lc_names_id)
+    variant_id = "http://www.loc.gov/mads/rdf/v1#variantLabel"
```
Comment (jamesgorrie) [inline inference/app/src/exact/lc_names.py:30]: This `#` isn't going to be read by the server, what is it intended to be doing?

Comment (harrisonpim) [reply inference/app/src/exact/lc_names.py:30]: the string is a key in the LoC API's (fairly ugly) response - see http://id.loc.gov/authorities/names/nb2015021801.json for an example

**Comment by jamesgorrie** [comment]
Am I right in saying we go:
`LC name ID => LOC => Wikidata ID => Wikidata Data`?

**Comment by jamesgorrie** [comment]
And we're not sanitising anything yet?

**Comment by harrisonpim** [comment]
yep, both correct for now. sanitising will come in later as we start making less direct, more inferred jumps from dataset to dataset

File: inference/app/src/exact/lc_names.py (hunk 1 above)
Comment (jamesgorrie) [reply inference/app/src/exact/lc_names.py:30]: `#variantLabel` <= this has though, is it doing anything?

Comment (harrisonpim) [reply inference/app/src/exact/lc_names.py:30]: we never send a request to `http://www.loc.gov/mads/rdf/v1#variantLabel`, it's literally just used as a key to get to the right bit of the json response

Comment (jamesgorrie) [reply inference/app/src/exact/lc_names.py:30]: Crumbles, not sure why I was getting so confused.

Comment (harrisonpim) [reply inference/app/src/exact/lc_names.py:30]: no worries! i have no idea why the api is organised that way - seems bound to create exactly this sort of confusion

**Comment by harrisonpim** [comment]
yeah, adding some of the [aiohttp](https://docs.aiohttp.org/en/stable/) magic that @jamieparkinson added to the feature-inferrer is definitely on the todo list, but I thought I'd get a simple skeleton working before complicating it with optimisation

**Pull request merged**